source/llassetgen/source/Packing.cpp

```diff
@@ -0,0 +1,90 @@
+#include <llassetgen/Packing.h>
+
+#include <algorithm>
+#include <tuple>
+
+namespace llassetgen {
+    Packing::Packing(Vec2<PackingSizeType> _atlasSize) : atlasSize{_atlasSize} {}
+
+    namespace impl_packing {
+        bool ShelfNextFitPacker::packNext(Vec2<PackingSizeType> rectSize) {
+            if (currentShelfSize.x + rectSize.x > packing.atlasSize.x) {
+                usedHeight += currentShelfSize.y;
+                if (rectSize.x > packing.atlasSize.x) {
+                    return false;
+                }
+
+                currentShelfSize = {0, 0};
```

Comment:
Could use `openNewShelf` here as well

Reply:
Using it in both methods was the very reason I factored that method out, but I must have forgotten to change it here. Fixed in the newest commit, thanks!